jdaviz/configs/specviz/plugins/line_analysis/line_analysis.py

```diff
@@ -332,7 +332,42 @@ def _uncertainty(result):
         for function in FUNCTIONS:
             # TODO: update specutils to allow ALL analysis to take regions and continuum so we
             # don't need these if statements
-            if function == "Equivalent Width":
+            if function == "Line Flux":
+                meta = self.dataset.selected_dc_item.meta
+                telescope = meta.get('TELESCOP',
+                                     meta.get('telescope',
+                                              meta.get('_primary_header', {}).get('TELESCOP', '')))
+                if telescope == 'JWST':
+                    # Perform integration in frequency space
+                    freq_spec = Spectrum1D(spectral_axis=spec_subtracted.spectral_axis.to(u.Hz),
+                                           flux=spec_subtracted.flux)
+                    temp_result = analysis.line_flux(freq_spec)
+
+                    # Convert result to Watts/meter^2 if in the right units
+                    # (don't forget the angle if it was provided)
+                    flux_unit = spec_subtracted.flux.unit.decompose()
+                    flux_unit_decompose = set(unit**power for unit, power in zip(flux_unit.bases,
+                                                                                 flux_unit.powers))
+                    if flux_unit_decompose == {u.Unit("1 / s2"), u.Unit("1 / rad2"), u.Unit("kg")}:
+                        # Multiply by PIXAR_SR if available
+                        viewer = self.app.get_viewer('spectrum-viewer')
+                        pixar_sr = meta.get('_primary_header', {}).get('PIXAR_SR', '')
+                        if (hasattr(viewer.state, 'function') and pixar_sr and
+                                # We can rely on the viewer's function because Cubeviz only allows
+                                # one cube per instance. If this changes, this loop will need to be
+                                # modified to look at the specific data entry's collapse function
+                                viewer.state.function.lower() in ('minimum', 'maxium', 'sum')):
+                            temp_result = temp_result * (float(pixar_sr) * u.Unit('sr'))
+                            # Multiplying by PIXAR_SR removes the steradian
+                            temp_result = temp_result.to(u.Unit('W/m2'))
+                        else:
+                            # If PIXAR_SR wasn't provided, keep the steradian unit
```

Comment:
but this `else` also catches cases where the viewer collapse function is something other than min/max/sum, right?  Is that expected behavior for the user or would that be confusing?

Reply:
The path here is super confusing (see https://github.com/spacetelescope/jdaviz/pull/1564#issuecomment-1213716389), so let me try to explain it and let's see if it makes sense, or if I confused myself 😅 

If the Flux units are in surface brightness (e.g. has the steradians in it), but `PIXAR_SR` was not provided, the units should ALWAYS be in W/(m2 sr). It doesn't matter what the collapse function is. The conversion to W/m2 ONLY happens when `PIXAR_SR` exists, AND the function is either min, max, or sum. 

To directly answer your question, if the function is mean/median and the flux is in surface brightness, then the units SHOULD NOT be converted to W/m2, instead SHOULD keep the steradian. Likewise, if the collapse function is min/max/sum and `PIXAR_SR` was NOT provided, then likewise the units should keep the steradian.

How'd I do?